ports/stm32/machine_i2s.c

```diff
@@ -566,6 +565,21 @@ STATIC bool i2s_init(machine_i2s_obj_t *self) {
         const pin_af_obj_t *af = pin_find_af(self->sd, AF_FN_I2S, self->i2s_id);
         GPIO_InitStructure.Alternate = (uint8_t)af->idx;
         HAL_GPIO_Init(self->sd->gpio, &GPIO_InitStructure);
+
+        #if defined(STM32H7)
```

Comment:
Is there any other way to code this besides putting a processor macro into the file.  e.g use a feature support macro in mpconfigboard.h files?   Sometimes I find that processor macros seem to grow in micropython files, which tend to undermine code comprehension.  Not a big deal, and more of a question on my part to learn the best practice to follow in micropython.

Reply:
I agree these things grow, but it's hard to make it clean.  The HAL is supposed to help there, and it does to some extent.

What's going on here is that the H7 (and maybe G4?) support SDI and SDO pins, and they can be swapped.  So instead of `#if defined(STM32H7)` it could be `#if MCU_SUPPORTS_SDI_SDO` (probably use a better name than that).  Does that sound reasonable?

Reply:
Thanks for explaining the issue.  Maybe just leave it as-is, and then consider refactoring if other processor variant(s) drive the use of additional macro logic?